### PR TITLE
Add storybook to test LandingPage component

### PR DESCRIPTION
**Pull request by juliandescottes**
I started a storybook integration but faced two issues so let's discuss.

1. had to move the Redux related code out of the LandingApp component. Using storybook with Redux seems feasible but complicated, not sure it's worth the investment
2. feature.js would fail on Chrome when trying to export its methods (getValue is undefined). Not sure what's up with that, might be linked to the default webpack config used by storybook

cc @jasonLaster ; will try to investigate more the last issue mentioned above

### CODE REVIEW

**Comment by jasonLaster** [comment]
Yup. That sounds frustrating.

I think the key is to just show the landing page with storybook.

We should pull feature out of the component and pass the values in as props
so it's testable.

Best thing to do with storybook is to imagine your writing a test

**Comment by juliandescottes** [comment]
I pulled the dependency on devtools-config/getValue out of the landing page component. Let me know what you think. 

We could add a "LaunchpadApp" intermediary between index.js and LandingPage.js, that would basically just call connect on the LandingPage component and pass the props, but I feel like it'd be overkill here.

File: packages/devtools-launchpad/src/components/LandingPage.js [hunk 1]
```diff
@@ -154,6 +152,4 @@ const LandingPage = React.createClass({
   }
 });
 
-module.exports = connect(
-  state => ({ tabs: getTabs(state) })
-)(LandingPage);
+module.exports = LandingPage;
```
Comment (jasonLaster) [inline packages/devtools-launchpad/src/components/LandingPage.js]: You should be able to keep this here and reference the wrappedCompondnt or something like that

**Comment by juliandescottes** [comment]
> I pulled the dependency on devtools-config/getValue out of the landing page component. Let me know what you think.

Would have been nice if I actually had pushed my changes ... Just updated the PR, sorry for pinging you too early.

File: packages/devtools-launchpad/src/index.js [hunk 2]
```diff
@@ -104,7 +105,19 @@ function bootstrap(React, ReactDOM, App, appActions, appStore) {
   }
 
   const { store, actions, LandingPage } = initApp();
-  renderRoot(React, ReactDOM, LandingPage, store);
+
+  let landingPage = connect(
```
Comment (jasonLaster) [inline packages/devtools-launchpad/src/index.js]: this should still be in the `LandingPage` component.

you can do something like 
`const { wrappedComponent: LandingPage } = require("components/LandingPage")`

Comment (juliandescottes) [inline packages/devtools-launchpad/src/index.js]: Thanks! Indeed it does fix my issue with connect!

**Comment by juliandescottes** [comment]
Adding back the dependency on "devtools-config" in LandingPage.js means the issue with feature.js on Chrome is back.

So if we land this as is, storybook only works in Firefox. I'll take a look at the generated bundle later to see if I can understand what's wrong here.

**Comment by juliandescottes** [comment]
@jasonLaster sorry about the delay, it got a bit busy during the previous days. 
I added a LaunchpadApp component to cleanup the LandingPage. It can now be tested properly on Firefox and Chrome.

**Comment by jasonLaster** [comment]
@juliandescottes I made some small tweaks so that it would be more react idiomatic.

Feel free to merge and or make other tweaks.

This is pretty great.

screen shot 2016-12-24 at 11 16 21 am

Lets also look at adding some automated testing here as well
https://getstorybook.io/docs/react-storybook/testing/react-ui-testing

The biggest advantage is that we don't run the risk of regressing on these stories

**Comment by juliandescottes** [comment]
Looks great @jasonLaster thanks! Merging.

**Comment by nchevobbe** [comment]
@juliandescottes do you have things to do here or can we merge the PR ?

**Comment by juliandescottes** [comment]
@nchevobbe no sorry ... pretty sure I clicked on merge, but looks like I missed the button ...

**Comment by nchevobbe** [comment]
No harms, the change looks great and will make it easy to hack on the landing page

**Pull request merged**